app/controllers/bills_controller.rb

```diff
@@ -133,4 +147,25 @@ def handle_error_update
   def new_bill
     @new_bill = Bill.new
   end
+
+  def cancel_bill
+    @bill_current.update!(status: Bill.statuses[:Canceled])
+  end
+
+  def handle_successful_update
+    @bill_current.bill_details.each do |bill_detail|
+      product_detail = bill_detail.product_detail
+      product_detail.update!(
+        quantity: product_detail.quantity + bill_detail.quantity
+      )
+    end
+  end
+
+  def load_bill
+    @bill_current = Bill.find_by id: params[:id]
+    return if @bill_current
```

Comment:
```suggestion
    @bill = Bill.find_by id: params[:id]
    return if @bill
```
ri dc, thêm chữ current làm j =)) thừa